.github/actions/add-checks/action.yml

```diff
@@ -0,0 +1,40 @@
+name: 'Add Status Checks'
+description: 'Add status check on a specified head-sha'
+inputs:
+  name:
+    description: 'Name of status check'
+    required: true
+  status:
+    description: >-
+      Status of check with possible values of 'queued', 'in_progress', 'completed'
+    required: true
+  head-sha:
+    description: 'The commit ID to add the checks to'
+    required: true
+  conclusion:
+    description: >-
+      Conclusion of status check with possible values of 'action_required',
+      'cancelled', 'failure', 'neutral', 'success', 'skipped', 'stale', 'timed_out'.
+      This is only required if status is 'completed'.
+    required: false
+runs:
+  using: 'composite'
+  steps:
+    - uses: actions/github-script@v6
```

Comment:
Prefer referencing github actions through commit sha rather than version (see codeql workflow)

Reply:
Updated to use commit sha as in codeql workflow